src/mgmt/srv_target.c

```diff
@@ -336,6 +339,159 @@ cleanup_leftover_pools(bool zombie_only)
 	cleanup_dead_list(&dead_list, newborns_path);
 }
 
+static int
+tgt_recreate(uuid_t pool_uuid, daos_size_t scm_size, int tgt_nr, daos_size_t rdb_blob_sz)
+{
+	char			*pool_newborn_path = NULL;
+	char			*pool_path = NULL;
+	char			*rdb_path = NULL;
+	bool			 dummy_cancel_state = false;
+	int			 rc;
+	int			 fd;
+	struct stat		 statbuf;
+
+	D_ASSERT(bio_nvme_configured(SMD_DEV_TYPE_META));
+
+	/** generate path to the target directory */
+	rc = ds_mgmt_tgt_file(pool_uuid, NULL, NULL, &pool_path);
+	if (rc) {
+		D_ERROR("newborn path_gen failed for "DF_UUID": "DF_RC"\n",
+			DP_UUID(pool_uuid), DP_RC(rc));
+		goto out;
+	}
+
+	/** Skip recreation if directory already exists */
+	rc = stat(pool_path, &statbuf);
+	if ((rc == 0) && (statbuf.st_mode & S_IFDIR))
+		goto out;
+
+	/** create the pool directory under NEWBORNS */
+	rc = path_gen(pool_uuid, newborns_path, NULL,
+		      NULL, &pool_newborn_path);
+	if (rc) {
+		D_ERROR(DF_UUID": path_gen failed for: "DF_RC"\n",
+			DP_UUID(pool_uuid), DP_RC(rc));
+		goto out;
+	}
+	rc = mkdir(pool_newborn_path, 0700);
+	if (rc < 0 && errno != EEXIST) {
+		rc = daos_errno2der(errno);
+		D_ERROR("failed to created pool directory: "DF_RC"\n",
+			DP_RC(rc));
+		/* avoid tgt_destroy(), nothing to do */
+		D_FREE(pool_newborn_path);
+		goto out;
+	}
+
+	/** create VOS files */
+	rc = tgt_vos_preallocate_parallel(pool_uuid, scm_size, tgt_nr,
+					  &dummy_cancel_state);
+	if (rc) {
+		D_ERROR(DF_UUID": failed to create tgt vos files: "DF_RC"\n",
+			DP_UUID(pool_uuid), DP_RC(rc));
+		goto out;
+	}
+
+	if (rdb_blob_sz) {
+		rc = path_gen(pool_uuid, newborns_path, RDB_FILE"pool",
+		      NULL, &rdb_path);
+		if (rdb_path == NULL) {
+			D_ERROR(DF_UUID": cannot retrieve rdb file info: "DF_RC"\n",
+				DP_UUID(pool_uuid), DP_RC(rc));
+			rc = -DER_NONEXIST;
+			goto out;
+		}
+		fd = open(rdb_path, O_RDWR|O_CREAT, 0600);
+		if (fd < 0) {
+			rc = daos_errno2der(errno);
+			D_ERROR("failed to create/open the vos file %s:"DF_RC"\n",
+				rdb_path, DP_RC(rc));
+			goto out;
+		}
+		rc = fallocate(fd, 0, 0, rdb_blob_sz);
+		close(fd);
+		if (rc) {
+			rc = daos_errno2der(errno);
+			D_ERROR("fallocate on rdb file %s failed:"DF_RC"\n",
+				rdb_path, DP_RC(rc));
+			goto out;
+		}
+		D_FREE(rdb_path);
+	}
+
+	/** move away from NEWBORNS dir */
+	rc = rename(pool_newborn_path, pool_path);
+	if (rc < 0) {
+		rc = daos_errno2der(errno);
+		D_ERROR(DF_UUID": failed to rename pool directory: "DF_RC"\n",
+			DP_UUID(pool_uuid), DP_RC(rc));
+		goto out;
+	}
+
+	/** make sure the rename is persistent */
+	(void)dir_fsync(pool_path);
+
+out:
+	D_FREE(pool_newborn_path);
+	D_FREE(pool_path);
+
+	return rc;
+}
+
+static int
+recreate_pooltgts()
+{
+	struct smd_pool_info    *pool_info = NULL, *tmp;
```

Comment:
style - should define tmp on a separate line.

Reply:
Thanks fixed it.